pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -275,6 +294,25 @@ func GetCreateLaunchTemplateInput(options *amifamily.LaunchTemplate, ClusterIPFa
 			},
 		},
 	}
+	// Gate this specifically since the update to CapacityReservationPreference will opt od / spot launches out of open
+	// ODCRs, which is a breaking change from the pre-native ODCR support behavior.
+	if karpoptions.FromContext(ctx).FeatureGates.ReservedCapacity {
```

Comment:
I wonder if we should also make the lack of capacityReservations gate this behavior as well -- effectively, if you don't select on capacity reservations, then you will still maintain the open matching behavior

Reply:
Maybe, but one reason I don't love that is that we could be consuming from ODCRs that other NodeClasses are targetting. I think this could lead to more unexpected behavior than anything. I don't think this is a one-way door decision, and I'd rather see us start with the more conservative approach.

Reply:
Discussed offline: I was mainly concerned with how this evolves as we look forward to this feature becoming stable -- you're right it's not a one-way door decision, we could always add that later _when_ we decide to make that feature stable. I also think a reasonable option is giving folks enough time who have been relying on open matching with capacityReservationTerms (I don't expect there to be many by the time this feature is stable, if any at all) to migrate so that they don't get burned by the change in behavior